cryptography/src/ed25519.rs

```diff
@@ -125,6 +125,47 @@ impl Scheme for Ed25519 {
     }
 }
 
+impl Ed25519 {
+    /// Check each signature is valid for the given message and public key.
+    ///
+    /// Because namespace is prepended to message before signing, the namespace
+    /// provided here must match the namespace provided during signing.
+    pub fn batch_verify(inputs: Vec<SignatureVerificationInput>) -> bool {
+        let mut batch = ed25519_consensus::batch::Verifier::new();
+        for input in inputs {
+            let item: ed25519_consensus::batch::Item = match input.try_into() {
+                Ok(item) => item,
+                Err(_) => return false,
+            };
+            batch.queue(item);
+        }
+        batch.verify(rand::thread_rng()).is_ok()
+    }
+}
+
+/// SignatureVerificationInput represents required information to verify a Signature.
+pub struct SignatureVerificationInput<'a> {
```

Comment:
When you make the change above to operate directly over the batch (instead of providing a vector), you should be able to get rid of this struct (just require these arguements directly in the `Ed25519Batch::add()` func).

Reply:
This is why I thought it would be possible to implement as a trait implementation.

We could have a new trait defined as `BatchScheme` or something and the `add` function should be generic enough for multiple schemes to implement:
```
fn add(
    namespace: Option<&'a [u8]>,
    message: &'a [u8],
    public_key: &'a PublicKey,
    signature: &'a Signature,
) -> Result<(), Error>
```

Or something like that. Once you get this PR in a good place, I could add this pretty easily (which we could also do for BLS).